frontend/src/pages/TrainingReports/components/EventCard.js

```diff
@@ -19,12 +19,20 @@ function EventCard({
   onRemoveSession,
 }) {
   const { user } = useContext(UserContext);
+
+  // Check region permissions.
   const hasEditPermissions = canEditOrCreateSessionReports(
     user,
     parseInt(event.regionId, DECIMAL_BASE),
   );
-  const isCollaborator = event.pocId && event.pocId.includes(user.id);
-  const canEditExisting = hasEditPermissions || (isCollaborator);
+
+  // Check if user has been assigned an event role.
+  const hasEditRole = (event.pocId && event.pocId.includes(user.id))
+    || (event.collaboratorIds && event.collaboratorIds.includes(user.id))
+    || (event.ownerId === user.id);
+
+  // Has region permissions and has been assigned an event role.
+  const canEditEvent = hasEditPermissions && hasEditRole;

```

Comment:
We need to amend this. The poc (someone who only has collaborator permissions) needs to be able to create a session.

Reply:
We'll probably want a FE test verifying that a collaborator sees the "create session" button as well

Reply:
Same user also can't create a session if they visit the url directly, so the backend permissions are also wrong

Reply:
The original chart that we're basing all this on is here:

https://app.mural.co/t/adhoccorporateworkspace2583/m/adhoccorporateworkspace2583/1683896571063/bc580f5a6ffa94c447fd8ad38f5535b5c8813ab6?invited=true&sender=u775286a3a84f5c5204174211